ksiren/src/test/java/com/brightspace/ksiren/KSirenRequestBuilderTest.kt

```diff
@@ -23,29 +25,40 @@ import kotlin.test.assertTrue
  */
 class KSirenRequestBuilderTest {
 
-    fun returnAction(method: String): Action {
-        return Action("test", listOf(), method, "http://www.example.com", "title", ContentType.FORM,
-                listOf(Field("testParam", listOf(), "text", null)))
-    }
+	private fun returnAction(method: String, contentType: ContentType): Action {
+		return Action(
+			"test",
+			listOf(),
+			method,
+			"http://www.example.com",
+			"title",
+			contentType,
+			listOf(Field("testParam", listOf(), "text", null)))
+	}
 
-    @Test
-    fun testGetBuilder() {
-        val action: Action = returnAction("GET")
-        val requestBuilder: KSirenRequestBuilder<Request> = KSirenOkhttp3RequestBuilder(action)
-        requestBuilder.addFieldValue("testParam", "testValue")
-        assertEquals("http://www.example.com/?testParam=testValue", requestBuilder.build().url().toString())
-    }
+	@Test
+	fun testGetBuilder() {
+		val action: Action = returnAction("GET", ContentType.FORM)
+		val requestBuilder: KSirenRequestBuilder<Request> = KSirenOkhttp3RequestBuilder(action)
+		requestBuilder.addFieldValue("testParam", "testValue")
+		assertEquals("http://www.example.com/?testParam=testValue", requestBuilder.build().url().toString())
+	}
 
-    @Test
-    fun testPostBuilder() {
-        val action: Action = returnAction("POST")
-        val requestBuilder: KSirenRequestBuilder<Request> = KSirenOkhttp3RequestBuilder(action)
-        requestBuilder.addFieldValue("testParam", "testValue")
-        requestBuilder.build()
+	@TestFactory
+	fun testPostBuilder() = listOf(
+		ContentType.FORM,
+		ContentType.JSON
+	).map { contentType ->
+			DynamicTest.dynamicTest("when Content-Type is $contentType") {
+				val action = returnAction("POST", contentType)
+				val requestBuilder: KSirenRequestBuilder<Request> = KSirenOkhttp3RequestBuilder(action)
+				requestBuilder.addFieldValue("testParam", "testValue")
```

Comment:
Would this be a place to add _interesting_ parameters to validate escaping is working properly? Or am I misunderstanding these tests?

Reply:
Yes, but also, there isn't a good check that's what's added as serialized correctly - marked as a TODO to get early feedback on the approach.

Reply:
I guess it's counting on no exceptions being thrown on the builder. Would it be worthwhile to parse the request that's built?

It would be nice peace of mind to have a failing test based on the old implementation due to unescaped characters that passes with your changes, but maybe that's wishful thinking.

Reply:
I have tests on the method that this will call to serialize-to-JSON now that it's in the right place, and yeah, it would be nice if these tests were somewhat better than "doesn't explode" - future work

Reply:
I have tests on the method that this will call when serializing JSON request bodies now that it's in the right place, and yeah, it would be nice if these tests were somewhat better than "doesn't explode" - future work